chalk-integration/src/program.rs

```diff
@@ -392,4 +392,8 @@ impl RustIrDatabase<ChalkIr> for Program {
     fn interner(&self) -> &ChalkIr {
         &ChalkIr
     }
+
+    fn is_object_safe(&self, trait_id: TraitId<ChalkIr>) -> bool {
+        self.trait_datum(trait_id).is_object_safe_trait()
```

Comment:
This should just be stored in  `Program`.

Reply:
Should I use `HashSet`, or is there a specific kind of set that should be used (`BTreeMap` of booleans)?